### PR TITLE
Allow to save fused point cloud in colmap format when using command line

### PR DESCRIPTION
**Pull request by boitumeloruf**
* Adjusted src/base/reconstruction.cc to import point cloud from std::vector<PlyPoints>
* Adjusted src/exe/colmap.cc to allow saving in colmap format when calling stereo_fusion from command line

See Issue https://github.com/colmap/colmap/issues/738

### CODE REVIEW

File: src/base/reconstruction.cc [hunk 1]
```diff
@@ -864,6 +864,16 @@ void Reconstruction::ImportPLY(const std::string& path) {
   }
 }
 
+void Reconstruction::ImportPLY(const std::vector<PlyPoint> &plyPoints)
```
Comment (ahojnnes) [inline src/base/reconstruction.cc:867]: `const std::vector<PlyPoint>& ply_points`

File: src/base/reconstruction.cc [hunk 2]
```diff
@@ -864,6 +864,16 @@ void Reconstruction::ImportPLY(const std::string& path) {
...
+{
+  points3D_.clear();
+  points3D_.reserve(plyPoints.size());
+  for (const auto& plyPoint : plyPoints) {
```
Comment (ahojnnes) [inline src/base/reconstruction.cc:871]: ply_point

File: src/base/reconstruction.h [hunk 3]
```diff
@@ -266,6 +266,7 @@ class Reconstruction {
   // Import from other data formats. Note that these import functions are
   // only intended for visualization of data and usable for reconstruction.
   void ImportPLY(const std::string& path);
+  void ImportPLY(const std::vector<PlyPoint>& plyPoints);
```
Comment (ahojnnes) [inline src/base/reconstruction.h:269]: ply_point

File: src/exe/colmap.cc [hunk 4]
```diff
@@ -300,10 +303,30 @@ int RunStereoFuser(int argc, char** argv) {
...
   fuser.Wait();
 
-  std::cout << "Writing output: " << output_path << std::endl;
-  WriteBinaryPlyPoints(output_path, fuser.GetFusedPoints());
-  mvs::WritePointsVisibility(output_path + ".vis",
-                             fuser.GetFusedPointsVisibility());
+  Reconstruction reconstruction;
+
+  // read data from sparse reconstruction
+  if (workspace_format == "colmap" )
```
Comment (ahojnnes) [inline src/exe/colmap.cc:309]: Remove space and wrap if body in {}

File: src/exe/colmap.cc [hunk 5]
```diff
@@ -300,10 +303,30 @@ int RunStereoFuser(int argc, char** argv) {
...
+  reconstruction.ImportPLY(fuser.GetFusedPoints());
+
+  // write output
+  StringToLower(&output_type);
+  if (output_type == "bin") {
+    reconstruction.WriteBinary(output_path);
+  } else if (output_type == "txt") {
+    reconstruction.WriteText(output_path);
+  } else if (output_type == "ply") {
+    std::cout << "Writing output: " << output_path << std::endl;
```
Comment (ahojnnes) [inline src/exe/colmap.cc:322]: Extract this to before the if

Comment (boitumeloruf) [reply src/exe/colmap.cc]: I didn't which line you meant. I moved the 'std::cout ...' before the if.

File: src/exe/colmap.cc [hunk 6]
```diff
@@ -266,6 +266,7 @@ int RunStereoFuser(int argc, char** argv) {
   std::string workspace_format = "COLMAP";
   std::string pmvs_option_name = "option-all";
   std::string output_path;
+  std::string output_type;
```
Comment (ahojnnes) [inline src/exe/colmap.cc:269]: Can we make the previous "ply" the default output type? This also requires to change to AddDefaultOption below. Thanks.

Comment (boitumeloruf) [reply src/exe/colmap.cc]: no problem

**Pull request merged**